packages/expansion-hub/src/internal/ExpansionHub.ts

```diff
@@ -87,8 +88,43 @@ export class ExpansionHubInternal implements ExpansionHub {
         this.nativeRevHub.setResponseTimeoutMs(timeout);
     }
 
-    getADC(): Promise<number> {
-        return this.nativeRevHub.getADC();
+    async getAnalogInput(channel: number): Promise<number> {
+        return await this.nativeRevHub.getADC(channel, 0);
+    }
+
+    async getSystemCurrent(
+        systemType: SystemCurrentType = SystemCurrentType.Battery,
+    ): Promise<number> {
+        switch (systemType) {
+            case SystemCurrentType.I2C:
+                return await this.nativeRevHub.getADC(5, 0);
+            case SystemCurrentType.DigitalIO:
+                return await this.nativeRevHub.getADC(4, 0);
+            case SystemCurrentType.Servo:
+                return await this.nativeRevHub.getADC(6, 0);
+            case SystemCurrentType.Battery:
+                return await this.nativeRevHub.getADC(7, 0);
+        }
+        return -1;
+    }
+
+    async getMotorCurrent(motorChannel: number): Promise<number> {
+        if (motorChannel > 3 || motorChannel < 0) {
+            throw new Error(`Motor Channel ${motorChannel} is out of range`);
```

Comment:
We have a special error type for this, right?

Reply:
Yes, but not in this PR. Now that the errors PR is merged into main, I can use the proper error type.